[2024-05-14T04:51:27.391+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-05-14T04:51:27.413+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: json_to_csv.students_transform scheduled__2024-05-14T04:48:00+00:00 [queued]>
[2024-05-14T04:51:27.418+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: json_to_csv.students_transform scheduled__2024-05-14T04:48:00+00:00 [queued]>
[2024-05-14T04:51:27.419+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-05-14T04:51:27.425+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): students_transform> on 2024-05-14 04:48:00+00:00
[2024-05-14T04:51:27.437+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=1388) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-05-14T04:51:27.438+0000] {standard_task_runner.py:63} INFO - Started process 1391 to run task
[2024-05-14T04:51:27.438+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'json_to_csv', 'students_transform', 'scheduled__2024-05-14T04:48:00+00:00', '--job-id', '110', '--raw', '--subdir', 'DAGS_FOLDER/json_to_csv.py', '--cfg-path', '/tmp/tmpouj8z4a8']
[2024-05-14T04:51:27.441+0000] {standard_task_runner.py:91} INFO - Job 110: Subtask students_transform
[2024-05-14T04:51:27.516+0000] {task_command.py:426} INFO - Running <TaskInstance: json_to_csv.students_transform scheduled__2024-05-14T04:48:00+00:00 [running]> on host 7331c320d95e
[2024-05-14T04:51:27.592+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='json_to_csv' AIRFLOW_CTX_TASK_ID='students_transform' AIRFLOW_CTX_EXECUTION_DATE='2024-05-14T04:48:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-05-14T04:48:00+00:00'
[2024-05-14T04:51:27.594+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-05-14T04:51:27.595+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-05-14T04:51:27.596+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/json_to_csv.py", line 54, in transform
    o += ",".join([str(data[column]) for column in config["columns"]]) + "\n"
                       ~~~~^^^^^^^^
TypeError: list indices must be integers or slices, not str
[2024-05-14T04:51:27.603+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=json_to_csv, task_id=students_transform, run_id=scheduled__2024-05-14T04:48:00+00:00, execution_date=20240514T044800, start_date=20240514T045127, end_date=20240514T045127
[2024-05-14T04:51:27.611+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 110 for task students_transform (list indices must be integers or slices, not str; 1391)
[2024-05-14T04:51:27.619+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-05-14T04:51:27.641+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-05-14T04:51:27.644+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
